### ISSUE TITLE
Use &[T] as parameter_ident for arrays

### ISSUE THREAD

**Issue by bjorn3** [oxidecomputer/typify#364]
Currently `&Vec<T>` is used which still requires heap allocation. For strings `&str` is already used.

**Comment by ahl** [comment]
That would be an improvement; are there particular scenarios you have in mind or a particular JSON schema?

**Comment by bjorn3** [comment]
Yeah, the Zulip api schema. Several api calls have a list as parameter. For example https://github.com/zulip/zulip/blob/113ac6eb98ffe192580ba0e25a7958346daeebf8/zerver/openapi/zulip.yaml#L5767

**Comment by ahl** [comment]
I see. Can you show an example of the code you'd hope to write that would consume the generated interface?
